modules/k3s/k3s_image_test.go

```diff
@@ -0,0 +1,109 @@
+package k3s_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/k3s"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+func Test_LoadImages(t *testing.T) {
+	ctx := context.Background()
+
+	k3sContainer, err := k3s.RunContainer(ctx,
+		testcontainers.WithImage("docker.io/rancher/k3s:v1.27.1-k3s1"),
+		testcontainers.WithWaitStrategy(wait.ForLog(".*Node controller sync successful.*").AsRegexp()),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Clean up the container
+	defer func() {
+		if err := k3sContainer.Terminate(ctx); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	kubeConfigYaml, err := k3sContainer.GetKubeConfig(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	restcfg, err := clientcmd.RESTConfigFromKubeConfig(kubeConfigYaml)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	k8s, err := kubernetes.NewForConfig(restcfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	provider, err := testcontainers.ProviderDocker.GetProvider()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// ensure nginx image is available locally
+	err = provider.PullImage(context.Background(), "nginx")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("Test load image not available", func(t *testing.T) {
+		err = k3sContainer.LoadImages(context.Background(), "nginx", "fake.registry/fake:non-existing")
+		if err == nil {
+			t.Fatal("should had failed")
+		}
+	})
+
+	t.Run("Test load image in cluster", func(t *testing.T) {
+		err = k3sContainer.LoadImages(context.Background(), "nginx")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		pod := &corev1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-pod",
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:            "nginx",
+						Image:           "nginx",
+						ImagePullPolicy: corev1.PullNever, // use image only if already present
+					},
+				},
+			},
+		}
+
+		_, err = k8s.CoreV1().Pods("default").Create(context.Background(), pod, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		time.Sleep(1 * time.Second)
+		pod, err = k8s.CoreV1().Pods("default").Get(context.Background(), "test-pod", metav1.GetOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if pod.Status.ContainerStatuses[0].State.Waiting.Reason == "ErrImageNeverPull" {
```

Comment:
The tests say there is a panic here 🤔 Could you take a look 🙏 ?

Reply:
It should be fixed now. However, the test is taking well over `400s` (> 6 minutes) to complete. I'm concerned this will make it fail due to timeouts.